frontend/src/store/credential.js

```diff
@@ -0,0 +1,218 @@
+  const cloudProfileStore = useCloudProfileStore()
+
+  const state = reactive({
+    secretBindings: {},
+    secrets: {},
+    quotas: {},
+  })
+
+  const isInitial = computed(() => {
+    return isEmpty(state.secretBindings)
+  })
+
+  function $reset () {
+    state.secretBindings = {}
+    state.secrets = {}
+    state.quotas = {}
+  }
+
+  async function fetchCredentials () {
+    const namespace = authzStore.namespace
+    try {
+      const { data: { secretBindings, secrets, quotas } } = await api.getCloudProviderCredentials(namespace)
+      setCredentials({ secretBindings, secrets, quotas })
+    } catch (err) {
+      $reset()
+      throw err
+    }
+  }
+
+  function setCredentials ({ secretBindings, secrets, quotas }) {
+    $reset()
+
+    secretBindings?.forEach(item => {
+      const key = namespaceNameKey(item.metadata)
+      set(state.secretBindings, [key], item)
+    })
+
+    secrets?.forEach(item => {
+      const key = namespaceNameKey(item.metadata)
+      set(state.secrets, [key], item)
+    })
+
+    quotas?.forEach(item => {
+      const key = namespaceNameKey(item.metadata)
+      set(state.quotas, [key], item)
+    })
+  }
+
+  const secretBindingList = computed(() => {
+    const secretBindings = Object.values(state.secretBindings)
+    return secretBindings.map(secretBinding => {
+      const secret = getSecret(secretBinding.secretRef)
+      Object.defineProperty(secretBinding, '_secret', {
+        value: secret,
+        configurable: true,
+        enumerable: false,
+      })
+
+      const quotas = (secretBinding.quotas || [])
+        .map(quota => get(state.quotas, namespaceNameKey(quota)))
+        .filter(Boolean)
+
+      Object.defineProperty(secretBinding, '_quotas', {
+        value: quotas,
+        configurable: true,
+        enumerable: false,
+      })
+
+      return secretBinding
+    })
+  })
+
+  const secretList = computed(() => {
+    return Object.values(state.secrets)
+  })
+
+  const quotaList = computed(() => {
+    return Object.values(state.quotas)
+  })
+
+  async function createCredential (params) {
+    const { data: { secretBinding, secret } } = await api.createCloudProviderCredential({ secretBinding: params.secretBinding, secret: params.secret })
+    updateCloudProviderCredential({ secretBinding, secret })
+    appStore.setSuccess('Cloud Provider credential created')
+  }
+
+  async function updateCredential (params) {
+    const { data: { secretBinding, secret } } = await api.updateCloudProviderCredential({ secretBinding: params.secretBinding, secret: params.secret })
+    updateCloudProviderCredential({ secretBinding, secret })
+    appStore.setSuccess('Cloud Provider credential updated')
+  }
+
+  async function deleteCredential (name) {
+    const namespace = authzStore.namespace
+
+    await api.deleteCloudProviderCredential({ namespace, name })
+    remove({ namespace, name })
+    appStore.setSuccess('Cloud Provider credential deleted')
```

Comment:
maybe an improvement for another PR: include the name of the affected object? (also for create and update)